cpp/models/ide_secir/parameters_io.cpp

```diff
@@ -0,0 +1,231 @@
+    }
+
+    // The first time we need is -4 * global_support_max.
+    Eigen::Index start_shift = -4 * global_support_max_index;
+    // The last time needed is dependent on the mean stay times in the Exposed compartment and in InfectedNoSymptoms
+    // before transition to InfectedSymptoms.
+    Eigen::Index last_time_index_needed = Eigen::Index(std::ceil(
+        (model.parameters.get<TransitionDistributions>()[Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)]
+             .get_mean(dt) +
+         model.parameters
+             .get<TransitionDistributions>()[Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)]
+             .get_mean(dt)) /
+        dt));
+    // Create TimeSeries with zeros.
+    for (Eigen::Index i = start_shift; i <= last_time_index_needed; i++) {
+        // Add time point.
+        model.m_transitions.add_time_point(
+            i * dt, TimeSeries<ScalarType>::Vector::Constant((int)InfectionTransition::Count, 0.));
+    }
+
+    //--- Calculate the flow InfectedNoSymptomsToInfectedSymptoms using the RKI data and store in the m_transitions object.---
+    ScalarType min_offset_needed = std::ceil(
+        model.m_transitions.get_time(0) -
+        1); //Need -1 if first time point is integer and just the floor value if not, therefore use ceil and -1
+    ScalarType max_offset_needed = std::ceil(model.m_transitions.get_last_time());
+
+    bool min_offset_needed_avail = false;
+    bool max_offset_needed_avail = false;
+    // Go through the entries of rki_data and check if date is needed for calculation. Confirmed cases are scaled.
+    // Define dumy variables to store the first and the last index of the TimeSeries where the considered entry of rki_data is potentially needed.
+    Eigen::Index idx_needed_first = 0;
+    Eigen::Index idx_needed_last  = 0;
+    ScalarType time_idx           = 0;
+    for (auto&& entry : rki_data) {
+        int offset = get_offset_in_days(entry.date, date);
+        if (!(offset < min_offset_needed) && !(offset > max_offset_needed)) {
+            if (offset == min_offset_needed) {
+                min_offset_needed_avail = true;
+            }
+            // Smallest index for which the entry is needed is when offset=std::ceil(time_idx).
+            idx_needed_first =
+                Eigen::Index(std::max(std::floor((offset - model.m_transitions.get_time(0) - 1) / dt), 0.));
+            // Biggest index for which the entry is needed is when offset=std::floor(time_idx-dt).
+            idx_needed_last = Eigen::Index(std::min(std::ceil((offset - model.m_transitions.get_time(0) + 1) / dt),
+                                                    double(model.m_transitions.get_num_time_points() - 1)));
+
+            for (Eigen::Index i = idx_needed_first; i <= idx_needed_last; i++) {
+                time_idx = model.m_transitions.get_time(i);
+                if (offset == int(std::floor(time_idx))) {
```

Comment:
For the interpolation of the data: This is the same scheme as in the LCT model right? I think it would make sense to define a separate function and use it both in the LCT and IDE model. Then it could easily be used for the ODE model as well and code duplication would be avoided.

Reply:
The concept for calculating a value for num_confirmed at a non-integer time point is the same (simple linear interpolation). However, the values are processed differently (compartments vs flows that are calculated) and the calculation methods do not match. Since we have to go through the entries in a while loop, it also makes no sense to put the linear interpolation in a function, so I see no code duplication here or no way to generalize the concepts.

Reply:
Okay lets move this to a new issue.